plugins/Lv2Effect/Lv2Effect.cpp

```diff
@@ -109,8 +110,13 @@ extern "C"
 PLUGIN_EXPORT Plugin *lmms_plugin_main(Model *_parent, void *_data)
 {
 	using KeyType = Plugin::Descriptor::SubPluginFeatures::Key;
-	auto eff = new Lv2Effect(_parent, static_cast<const KeyType*>(_data));
-	if (!eff->isValid()) { delete eff; eff = nullptr; }
+	Lv2Effect* eff;
+	try {
+		eff = new Lv2Effect(_parent, static_cast<const KeyType*>(_data));
+	} catch (const std::runtime_error& e) {
+		qCritical() << e.what();
+		eff = nullptr;
+	}
 	return eff;
```

Comment:
I think it would be simpler to return directly in each case, rather than using a local variable (which is forced to be declared without being initialised).
```suggestion
	try {
		return new Lv2Effect(_parent, static_cast<const KeyType*>(_data));
	} catch (const std::runtime_error& e) {
		qCritical() << e.what();
		return nullptr;
	}
```